### ISSUE TITLE
Mouse clicks causing crashes when `number`/`relativenumber`  is used with a custom `statuscolumn`

### ISSUE THREAD

**Issue by OXY2DEV** [neovim/neovim#29390]
### Problem

If you set `number` and/or `relativenumber` to true then change the value of `statuscolumn` to something other than the default value strange empty columns can be seen. When clicked on them Neovim crashes.

![Screenshot_2024-06-18-13-42-36-278_com termux-edit](https://github.com/neovim/neovim/assets/122956967/d414ea04-27fd-4fdd-ad31-da3264060643)
![Screenshot_2024-06-18-13-42-56-060_com termux-edit](https://github.com/neovim/neovim/assets/122956967/bf62b6ba-78ed-4a77-bade-9dd903553f55)
![Screenshot_2024-06-18-13-43-28-247_com termux-edit](https://github.com/neovim/neovim/assets/122956967/d05ee2c7-0023-48b8-b25d-4dec24c31c13)
![Screenshot_2024-06-18-13-43-41-114_com termux-edit](https://github.com/neovim/neovim/assets/122956967/989374e1-ba6a-41a2-8c36-d1cae74a31cd)

### Steps to reproduce

1. Open a `clean` neovim session with `nvim --clean`.
2. You need to set the following options(either through the cmdline or your `init.lua` if you aren't running with the `--clean` flag.
```
vim.o.number = true;
-- vim.o.relativenumber = true;
-- Setting either one of them will work

vim.o.numberwidth = 10;
-- Optional, for the sake of demonstration a large value is used

vim.o.statuscolumn = "Test";
-- Can't be nil, " " as they become the default statuscolumn, has to be shorter than the numberwidth
```
3. You should see something like `Test      `. Click on the empty spaces after `Test`.

This should cause neovim to crash.

### Expected behavior

The clicks shouldn't be registered and nothing should occur.

### Neovim version (nvim -v)

NVIM v0.10.0 Build type: RelWithDebInfo LuaJIT 2.1.0-beta3

### Vim (not Nvim) behaves the same?

Not reproducible as vim doesn't have the `statuscolumn` option(at least I couldn't set it).

### Operating system/version

Android 13 aarch64(I am running inside Termux and proot distro isn't being used)

### Terminal name/version

Termux 0.118.0(from Github releases)

### $TERM environment variable

xterm-256color

### Installation

I have built from source, luarocks has been used to handle dependencies(as the other methods don't work).

**Comment by OXY2DEV** [comment]
> The 'statuscolumn' intentionally follows 'numberwidth', which is documented. I don't think that's strange?

Yeah, when I first saw the issue occur I had a custom statuscolumn(which was also 4 columns wide) set. So, when I saw a window where the statuscolumn should be empty also taking 4 columns I thought maybe Neovim made all the statuscolumn take the same amount of width. Which is why I said that.

I will go edit my comment.

**Comment by luukvbaal** [comment]
Returning early when `do_mouse()` when `click_col  wp->w_statuscol_click_defs_size()` seems appropriate. I.e. still consider it a statuscolumn click for other (non)side effects, but do not try to call the click func for area that was not part of the generated statuscolumn but padded instead.
